src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -866,6 +885,13 @@ - (void)invalidateCASESessionForNode:(chip::NodeId)nodeID;
             self->_cppCommissioner->GetPeerScopedId(nodeID), chip::MakeOptional(chip::Transport::SecureSession::Type::kCASE));
     });
 }
+
+- (void)setDeviceDiscoveryDelegate:(id<MTRDeviceDiscoveryDelegate>)delegate queue:(dispatch_queue_t)queue
+{
+    VerifyOrReturn([self checkIsRunning]);
```

Comment:
Silent no-op is not a great API....

Reply:
I removed all discovery and ethernet pairing changes because they are useful for test but not a mandatory feature for Matter certification.